### PR TITLE
feat(call-history): update the callhistory when the user reads missed calls

### PR DESCRIPTION
**Pull request by sokn-sys**
# COMPLETES # https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-452240

## This pull request addresses

Created the new method to updateMissedCalls(), to update the missed calls. This method will post a request containing the session ID and end time of unread missed calls, along with the Webex access token included in the headers.​
After the post request, when the "**isRead**" item is added for a given user in the userSession response, Janus will trigger the **event:janus_user_viewed_session** mercury event to notify corresponding clients.

## by making the following changes
In the MSFT client side, when a user clicks on any call history record or the voicemail sub-tab, and an event is triggered and the userSession response is updated, the missed call notification badge/counter should dismiss.

https://github.com/webex/webex-js-sdk/assets/58249052/990c58aa-47ac-48db-8af4-85903b856681

sdkTest
sdk1
sdk2
client3

sdk_readStatus

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested

< ENUMERATE TESTS PERFORMED, WHETHER MANUAL OR AUTOMATED >

### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [x] I discussed changes with code owners prior to submitting this pull request

- [x] I have not skipped any automated checks
- [x] All existing and new tests passed
- [x] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: packages/calling/src/CallHistory/types.ts
Comment (Shreyas281299) [inline packages/calling/src/CallHistory/types.ts:53]: Please add a description to the `updateMissedCalls` API, similar to getCallHistoryData.

**Review by Shreyas281299 — CHANGES_REQUESTED** [review]
No big changes required. Code looks good.

File: packages/calling/src/CallHistory/types.ts
Comment (sokn-sys) [reply packages/calling/src/CallHistory/types.ts:64]: Done

File: packages/calling/src/CallHistory/constants.ts
Comment (sokn-sys) [reply packages/calling/src/CallHistory/constants.ts]: Done

File: packages/calling/src/CallHistory/CallHistory.test.ts
Comment (sokn-sys) [reply packages/calling/src/CallHistory/CallHistory.test.ts]: Done

File: packages/calling/src/CallHistory/CallHistory.test.ts
Comment (sokn-sys) [reply packages/calling/src/CallHistory/CallHistory.test.ts]: Done

File: packages/calling/src/CallHistory/CallHistory.test.ts
Comment (Shreyas281299) [inline packages/calling/src/CallHistory/CallHistory.test.ts:162]: Leave a line after each test.

**Review by Shreyas281299 — APPROVED** [review]

File: packages/calling/src/CallHistory/CallHistory.test.ts
Comment (sokn-sys) [reply packages/calling/src/CallHistory/CallHistory.test.ts:164]: done

File: packages/calling/src/CallHistory/CallHistory.test.ts
Comment (Kesari3008) [inline packages/calling/src/CallHistory/CallHistory.test.ts:173]: You can club these 2 expect statements into one: using toBeAclledOnceWith()

File: packages/calling/src/CallHistory/CallHistory.test.ts
Comment (Kesari3008) [inline packages/calling/src/CallHistory/CallHistory.test.ts:235]: same comment here as above

**Review by Kesari3008 — CHANGES_REQUESTED** [review]

File: packages/calling/src/CallHistory/CallHistory.ts
Comment (sokn-sys) [reply packages/calling/src/CallHistory/CallHistory.ts:182]: The data field contains only the statusCode. We parse the response data and construct the response details.

File: packages/calling/src/CallHistory/CallHistory.ts
Comment (sokn-sys) [reply packages/calling/src/CallHistory/CallHistory.ts:149]: Done

File: packages/calling/src/CallHistory/CallHistory.ts
Comment (sokn-sys) [reply packages/calling/src/CallHistory/CallHistory.ts]: Done

File: packages/calling/src/CallHistory/CallHistory.ts
Comment (sokn-sys) [reply packages/calling/src/CallHistory/CallHistory.ts]: Done

File: packages/calling/src/CallHistory/CallHistory.test.ts
Comment (sokn-sys) [reply packages/calling/src/CallHistory/CallHistory.test.ts]: Done

File: packages/calling/src/CallHistory/CallHistory.test.ts
Comment (sokn-sys) [reply packages/calling/src/CallHistory/CallHistory.test.ts:233]: Done

Comment (Kesari3008) [reply packages/calling/src/CallHistory/CallHistory.test.ts:233]: This is not addressed

Comment (Kesari3008) [reply packages/calling/src/CallHistory/CallHistory.test.ts:233]: Also since the second argument here is the same object every time, instead of having all the fields to be copied again and again. Create an object to store it's value and pass that here

File: packages/calling/src/CallHistory/types.ts
Comment (Kesari3008) [inline packages/calling/src/CallHistory/types.ts:54]: It should be update the read state not updates

**Review by Kesari3008 — APPROVED** [review]
Approving with minor comments. Please take care of them before merging. Also did you test after changing the filed name from data to readStatusMessage? Ensure to test thoroughly before merge

File: packages/calling/src/CallHistory/types.ts
Comment (sokn-sys) [reply packages/calling/src/CallHistory/types.ts:54]: Done

File: packages/calling/src/CallHistory/CallHistory.test.ts
Comment (sokn-sys) [reply packages/calling/src/CallHistory/CallHistory.test.ts]: Done

**Comment by sokn-sys** [comment]
> Approving with minor comments. Please take care of them before merging. Also did you test after changing the filed name from data to readStatusMessage? Ensure to test thoroughly before merge

 I've addressed the comments and tested thoroughly after changing the field name from "data" to "readStatusMessage." Everything is working as expected, and the tests are passing. We're good to merge

**Pull request merged**
